scalameter-core/src/main/java/org/scalameter/execution/invocation/instrumentation/MethodInvocationCounter.java

```diff
@@ -0,0 +1,65 @@
+package org.scalameter.execution.invocation.instrumentation;
+
+import java.util.concurrent.atomic.AtomicLongArray;
+
+
+/**
+ * INTERNAL API
+ *
+ * Stores method invocations in a AtomicLongArray.
+ */
+public class MethodInvocationCounter {
+    private static volatile boolean _initialized = false;
+    private static volatile boolean _started = false;
+    private static AtomicLongArray _counts;
+
+    /**
+     * Initializes count array.
+     *
+     * @param numMethods length of an array
+     */
+    public static synchronized void setup(int numMethods) {
+        _counts = new AtomicLongArray(numMethods);
+        _initialized = true;
+    }
+
+    /**
+     * Starts measuring of method invocations.
+     */
+    public static synchronized void start() {
+        if (_initialized && !_started) {
+            _started = true;
+        }
+    }
+
+    /**
+     * Stops measuring of method invocations.
+     */
+    public static synchronized void stop() {
+        if (_initialized && _started) {
+            _started = false;
+        }
+    }
```

Comment:
Sorry, didn't complete it - I wanted to say that if you have the sequence of states: UNINITIALIZED -> INITIALIZED -> STARTED -> STOPPED, maybe you want to model it explicitly like that, instead of using a pair of booleans. It would be less error-prone.